deleter/ec2.go

```diff
@@ -1154,12 +1168,106 @@ func (rd *EC2NatGatewayDeleter) DeleteResources(cfg *DeleteConfig) error {
 			return err
 		}
 
+	}
+
+	if cfg.DryRun {
+		return nil
+	}
+
+	// If we don't wait until nat gateways are deleted, EIP disassociation/release
+	// and customer gateway disassociation/deletion will fail
+	fmt.Println("Waiting for EC2 NAT Gateways to delete...")
+	deletedNGWs, aliveNGWs, err := rd.waitUntilNatGatewaysDeleted(ngws)
+	if !cfg.IgnoreErrors {
+		fmt.Printf("{\"error\": \"%s\"}", err)
+	}
+	if len(aliveNGWs) != 0 {
+		for _, ngw := range aliveNGWs {
+			fmt.Printf("Could not delete EC2 Nat Gateway %s in 5 minutes (state \"%s\")\n", *ngw.NatGatewayId, *ngw.State)
+		}
+	}
+	for _, ngw := range deletedNGWs {
 		fmt.Println(fmtStr, *ngw.NatGatewayId)
 	}
 
 	return nil
 }
 
+func (rd *EC2NatGatewayDeleter) waitUntilNatGatewaysDeleted(ngws []*ec2.NatGateway) (deletedNGWs, aliveNGWs []*ec2.NatGateway, err error) {
+	// Set timeout to 5 minutes
+	timeout := time.Now().Add(time.Duration(300) * time.Second)
```

Comment:
Adding a config file key for such a specific timeout seems excessive to me. A 5 minute timeout is should be more than enough right? Nat gateways usually don't take longer than a minute to delete, regardless of the number you've requested to be deleted.